src/services/resource-arn.ts

```diff
@@ -0,0 +1,75 @@
+import {HttpVerb, toHttpVerb} from "./http-verbs";
+
+export interface ResourceArn {
+  region: string
+  accountId: string,
+  apiId: string,
+  stage: string,
+  httpVerb: HttpVerb,
+  resource: string | null,
```

Comment:
silly question, can the resource ever be null?

Reply:
yep - [Input](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-lambda-authorizer-input.html) and [Output](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-lambda-authorizer-output.html) documentation specify it as an optional field (anything between `[` and `]` in input > ... > `methodArn` and output > ... > `Resource` counts as optional)